tests/test_mig_shared_url.py

```diff
@@ -0,0 +1,107 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
+# USA.
+#
+# --- END_HEADER ---
+#
+
+"""Unit tests for the migrid module pointed to in the filename"""
+
+import os
+import sys
+
+sys.path.append(os.path.join(os.path.dirname(os.path.abspath(__file__))))
+
+from tests.support import MigTestCase, FakeConfiguration, testmain
+from mig.shared.url import _get_site_urls, check_local_site_url
+
+ENC_URL = 'https%3A%2F%2Fsomewhere.org%2Fsub%0A'
+VALID_STATIC_SITE_URLS = ['', 'abc', 'abc.txt', '/', '/bla', '/bla#anchor',
+                          '/bla/', '/bla/#anchor', '/bla/bla', '/bla/bla/bla',
+                          '//bla//', './bla', './bla/', './bla/bla',
+                          './bla/bla/bla', 'logout.py', 'logout.py?bla=',
+                          '/cgi-sid/logout.py', '/cgi-sid/logout.py?bla=bla',
+                          '/cgi-sid/logout.py?return_url=%s' % ENC_URL,
+                          ]
+
+
+class BasicUrl(MigTestCase):
+    """Wrap unit tests for the corresponding module"""
+
+    def _generate_dynamic_site_urls(self, url_base_list):
+        """Simple helper to construct a list of similar URLs for each base url
+        in url_base_list.
+        """
+        site_urls = []
+        for url_base in url_base_list:
+            site_urls += ['%s' % url_base, '%s/' % url_base,
+                          '%s/wsgi-bin/home.py' % url_base,
+                          '%s/wsgi-bin/logout.py' % url_base,
+                          '%s/wsgi-bin/logout.py?return_url=' % url_base,
+                          '%s/wsgi-bin/logout.py?return_url=%s' % (url_base,
+                                                                   ENC_URL)
+                          ]
+        return site_urls
+
+    def test_valid_local_site_urls(self):
+        """Check known valid static and dynamic URLs"""
+        dummy_conf = FakeConfiguration(migserver_http_url='http://myfqdn.org',
+                                       migserver_https_url='https://myfqdn.org',
+                                       migserver_https_mig_cert_url='',
+                                       migserver_https_ext_cert_url='',
+                                       migserver_https_mig_oid_url='',
+                                       migserver_https_ext_oid_url='',
+                                       migserver_https_mig_oidc_url='',
+                                       migserver_https_ext_oidc_url='',
+                                       migserver_https_sid_url='',
+                                       migserver_public_url='',
+                                       migserver_public_alias_url='')
+        url_base_urls = _get_site_urls(dummy_conf)
+        valid_site_urls = VALID_STATIC_SITE_URLS
+        valid_site_urls += self._generate_dynamic_site_urls(url_base_urls)
```

Comment:
Ideally we would not generate test data like this - for the record, the only reason other tests do so is where they are ports from elsewhere in the tree. The reason being, it is _highly_ beneficial to be able to see the test data being fed in.

I'd prefer to see constants at the top level of the module that contain these lists with all their items listed out explicitly - if that's not deemed amenable, then let's at least generate these the module level and assign them to underscroed all upper case variables. That way we someone can much easier print out what the contents of these things is, and there is no benefit or necessity for these things to be generated within each test.